tensorflow/compiler/mlir/hlo/lib/Dialect/lhlo/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -462,6 +462,96 @@ class GatherOpConverter : public OpRewritePattern<GatherOp> {
   }
 };
 
+/// Converts PadOp to Affine nest form.
+/// Pseudocode:
+///   1. Fill `output` tensor with `padding_value`.
+///   2. Compute AffineMap for store into `output`.
+///      out_idx = edge_padding_low +
+///                operand_idx * (1 + interior_padding)
+///   3. Create nested loop from `operand` shape.
+///      3.1 load from `operand`.
+///      3.2 store into `output`.
+/// Limitation for now:
+///   interior_padding == 0 && edge_padding_* >= 0
+struct PadOpConverter : public OpRewritePattern<PadOp> {
+  using OpRewritePattern<PadOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(PadOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value operand = op.operand();
+    Value padding_value = op.padding_value();
+    Value output = op.output();
+
+    ShapedType operand_type = operand.getType().dyn_cast<ShapedType>();
+    ShapedType output_type = output.getType().dyn_cast<ShapedType>();
+    unsigned rank = operand_type.getRank();
+    assert(output_type.getRank() == rank);
```

Comment:
Is the verifier for this op forbidding unranked input?

Reply:
There is no verifier for `lmhlo.Pad` and I've made changes to assert if the operands `hasRank()`.

Reply:
If there is no verifier, then this assert could fire, couldn't it?

Reply:
Yes @joker-eph , it can and instead of asserting on it I've included a check which bails out on lowering instead by returning `failure()`.